hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -2498,6 +2482,9 @@ class Emit[C](
               cb.assign(stageName, stageName.concat("|").concat(dynamicID.asString.loadString(cb)))
             })
 
+          val encRes = cb.newLocal[Array[Array[Byte]]]("encRes")
+          val len = mb.newLocal[Int]("cda_result_length")
+          val ib = mb.newLocal[InputBuffer]("decode_ib")
```

Comment:
why sometimes class builder sometimes method builder?

Reply:
More direct: How do you pick when to use mb vs cb?

Reply:
These are aliased. Rewrote to use the more idiomatic "memoize" now and avoid explicit variable declarations.

The rule is basically to use the lowest-level builder that has the functionality you want. Module > Class > Method > Code, but codeBuilder doesn't duplicate the full functionality of everything above. So to create a field, use methodBuilder (it exists on MB and ClassB, not codeB).